libraries/AP_IRLock/AP_IRLock_I2C.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * AP_IRLock_I2C.cpp
+ *
+ * Based on AP_IRLock_PX4 by MLandes
+ *
+ * See: http://irlock.com/pages/serial-communication-protocol
+ */
+#include <AP_HAL/AP_HAL.h>
+#include "AP_IRLock_I2C.h"
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define IRLOCK_I2C_ADDRESS		0x54
+
+#define IRLOCK_SYNC			0xAA55AA55
+
+void AP_IRLock_I2C::init()
+{
+    dev = std::move(hal.i2c_mgr->get_device(1, IRLOCK_I2C_ADDRESS));
```

Comment:
I agree. IRLock currently doesn't have any parameters at all, and it being tied to one bus isn't the only problem with the driver. 
The aim of this PR is to get the driver in-tree with existing functionality, then we can improve on it. It should be converted to use a frontend-backend split like other drivers, and get parameters (at least for bus address).
The way it handles health status also needs fixing. The old driver always reported as healthy. I changed that to report healthy when it had received a target in the last 0.1s. That works, but MissionPlanner saw reports of errors from precision land when it went unhealthy (when it couldn't see a target).